### PR TITLE
Move some namespaces on new Metadata

### PR DESCRIPTION
**Pull request by loic425**
| Q               | A
| --------------- | -----
| Bug fix?        | no
| New feature?    | no
| BC breaks?      | no
| Deprecations?   | no
| Related tickets | 
| License         | MIT

### CODE REVIEW

**Review by oallain — APPROVED** [review]
LGFM, but I don't know how to validate that everything is good, since it's a big search/replace.

**Comment by loic425** [comment]
> LGFM, but I don't know how to validate that everything is good, since it's a big search/replace.

Yes, these classes have been introduced in 1.11 so, they don't need any bc-layer.

File: docs/resource_factories.md [hunk 1]
```diff
@@ -96,9 +96,7 @@ declare(strict_types=1);
 
 namespace App\Entity\Book;
 
-use Sylius\Component\Resource\Metadata\Create;
-use Sylius\Component\Resource\Metadata\AsResource;
-use Sylius\Component\Resource\Model\ResourceInterface;
+use Sylius\Component\Resource\Model\ResourceInterface;use Sylius\Resource\Metadata\AsResource;use Sylius\Resource\Metadata\Create;
```
Comment (diimpp) [inline docs/resource_factories.md]: Please split on multiline.

File: docs/resource_factories.md [hunk 2]
```diff
@@ -166,9 +164,7 @@ declare(strict_types=1);
...
```
Comment (diimpp) [inline docs/resource_factories.md]: Please split on multiline.

File: docs/resource_factories.md [hunk 3]
```diff
@@ -192,9 +188,7 @@ declare(strict_types=1);
 
 namespace App\Entity\Book;
 
-use App\Factory\BookFactory;use Sylius\Component\Resource\Metadata\Create;
-use Sylius\Component\Resource\Metadata\AsResource;
-use Sylius\Component\Resource\Model\ResourceInterface;
+use App\Factory\BookFactory;use Sylius\Component\Resource\Model\ResourceInterface;use Sylius\Resource\Metadata\AsResource;use Sylius\Resource\Metadata\Create;
```
Comment (diimpp) [inline docs/resource_factories.md]: Please split on multiline.

File: docs/resource_factories.md [hunk 4]
```diff
@@ -237,11 +231,7 @@ declare(strict_types=1);
 
 namespace App\Entity\Book;
 
-use App\Entity\Book;
-use App\Factory\BookFactory;
-use Sylius\Component\Resource\Metadata\Create;
-use Sylius\Component\Resource\Metadata\AsResource;
-use Sylius\Component\Resource\Model\ResourceInterface;
+use App\Factory\BookFactory;use Sylius\Resource\Metadata\AsResource;use Sylius\Resource\Metadata\Create;
```
Comment (diimpp) [inline docs/resource_factories.md]: Please split on multiline.

File: docs/redirect.md [hunk 5]
```diff
@@ -65,9 +65,6 @@ declare(strict_types=1);
 
 namespace App\Entity;
 
-use Sylius\Component\Resource\Metadata\Create;
-use Sylius\Component\Resource\Metadata\AsResource;
-use Sylius\Component\Resource\Metadata\Update;
 use Sylius\Component\Resource\Model\ResourceInterface;
```
Comment (diimpp) [inline docs/redirect.md:68]: Missing `Create`

```php
use Sylius\Component\Resource\Metadata\Create;
```

File: docs/responders.md [hunk 6]
```diff
@@ -40,8 +40,8 @@ As an example, we add a `foo` variable to the Twig template with `bar` as value.
 
 namespace App\Twig\Context\Factory;
 
-use phpDocumentor\Reflection\Types\Context;
-use Sylius\Component\Resource\Metadata\Operation;
+use Sylius\Resource\Context\Context;
```
Comment (diimpp) [inline docs/responders.md:43]: Please sort alphabetically.

File: src/Bundle/spec/Twig/Context/LegacyContextFactorySpec.php [hunk 7]
```diff
@@ -18,10 +18,10 @@
 use Sylius\Bundle\ResourceBundle\Controller\RequestConfiguration;
 use Sylius\Bundle\ResourceBundle\Twig\Context\LegacyContextFactory;
 use Sylius\Component\Resource\Metadata\MetadataInterface;
```
Comment (diimpp) [inline src/Bundle/spec/Twig/Context/LegacyContextFactorySpec.php:20]: Shouldn't it be moved as well if other `Metadata` interfaces moved?

File: src/Component/Symfony/Routing/Factory/OperationRouteFactory.php [hunk 8]
```diff
@@ -14,10 +14,10 @@
 namespace Sylius\Component\Resource\Symfony\Routing\Factory;
 
 use Gedmo\Sluggable\Util\Urlizer;
-use Sylius\Component\Resource\Metadata\HttpOperation;
 use Sylius\Component\Resource\Metadata\MetadataInterface;
```
Comment (diimpp) [inline src/Component/Symfony/Routing/Factory/OperationRouteFactory.php:17]: Shouldn't it be moved as well?

File: src/Component/Symfony/Routing/Factory/OperationRouteFactoryInterface.php [hunk 9]
```diff
@@ -13,9 +13,9 @@
 
 namespace Sylius\Component\Resource\Symfony\Routing\Factory;
 
-use Sylius\Component\Resource\Metadata\HttpOperation;
 use Sylius\Component\Resource\Metadata\MetadataInterface;
```
Comment (diimpp) [inline src/Component/Symfony/Routing/Factory/OperationRouteFactoryInterface.php:16]: Shouldn't it be moved as well?

**Review by diimpp — CHANGES_REQUESTED** [review]
I've browsed around 40%, will finish later this day

File: docs/redirect.md (hunk 5 above)
Comment (diimpp) [reply docs/redirect.md:68]: And
```patch
--- #[Resource]
+++ #[AsResource]
```

Comment (loic425) [reply docs/redirect.md:68]: Fixed. I'm always impressed by your reviews. Thx.

File: docs/responders.md (hunk 6 above)
Comment (loic425) [reply docs/responders.md:43]: done

File: src/Bundle/spec/Twig/Context/LegacyContextFactorySpec.php (hunk 7 above)
Comment (loic425) [reply src/Bundle/spec/Twig/Context/LegacyContextFactorySpec.php:20]: It should but it has been introduced years ago, and should be part of another PR with a bc layer.

File: src/Bundle/Resources/config/services/metadata.xml [hunk 10]
```diff
@@ -17,15 +17,15 @@
             <tag name="cache.pool" />
         </service>
 
-        <service id="sylius.resource_metadata_collection.factory.attributes" class="Sylius\Component\Resource\Metadata\Resource\Factory\AttributesResourceMetadataCollectionFactory">
+        <service id="sylius.resource_metadata_collection.factory.attributes" class="Sylius\Resource\Metadata\Resource\Factory\AttributesResourceMetadataCollectionFactory">
```
Comment (diimpp) [inline src/Bundle/Resources/config/services/metadata.xml:20]: Not related to current PR, but this service has two aliases defined below.

image

I would probably make `id=Sylius\Resource\Metadata\Resource\Factory\ResourceMetadataCollectionFactoryInterface` main service and this `sylius.resource_metadata_collection.factory.attributes` as alias and deprecate other alias or all aliases.

File: src/Component/spec/Symfony/Routing/Factory/CollectionOperationRoutePathFactorySpec.php [hunk 11]
```diff
@@ -47,7 +46,7 @@ function it_generates_route_path_for_index_operations_with_custom_short_name():
 
     function it_generates_route_path_for_api_post_operations(): void
     {
-        $operation = new Api\GetCollection();
+        $operation = new \Sylius\Resource\Metadata\Api\GetCollection();
```
Comment (diimpp) [inline src/Component/spec/Symfony/Routing/Factory/CollectionOperationRoutePathFactorySpec.php]: Please move definition to `use` list.

File: src/Component/spec/Symfony/Routing/Factory/CreateOperationRoutePathFactorySpec.php [hunk 12]
```diff
@@ -47,7 +46,7 @@ function it_generates_route_path_for_create_operations_with_custom_short_name():
 
     function it_generates_route_path_for_api_post_operations(): void
     {
-        $operation = new Api\Post();
+        $operation = new \Sylius\Resource\Metadata\Api\Post();
```
Comment (diimpp) [inline src/Component/spec/Symfony/Routing/Factory/CreateOperationRoutePathFactorySpec.php]: Please move definition to `use` list.

File: src/Component/spec/Symfony/Routing/Factory/DeleteOperationRoutePathFactorySpec.php [hunk 13]
```diff
@@ -55,14 +54,14 @@ function it_generates_route_path_for_update_operations_with_custom_short_name():
 
     function it_generates_route_path_for_api_delete_operations(): void
     {
-        $operation = new Api\Delete();
+        $operation = new \Sylius\Resource\Metadata\Api\Delete();
```
Comment (diimpp) [inline src/Component/spec/Symfony/Routing/Factory/DeleteOperationRoutePathFactorySpec.php]: Please move definition to `use` list.

File: src/Component/spec/Symfony/Routing/Factory/DeleteOperationRoutePathFactorySpec.php [hunk 14]
```diff
@@ -55,14 +54,14 @@ function it_generates_route_path_for_update_operations_with_custom_short_name():
...
 
         $this->createRoutePath($operation, '/dummies')->shouldReturn('/dummies/{id}');
     }
 
     function it_generates_route_path_for_api_delete_operations_with_custom_short_name(): void
     {
-        $operation = new Api\Delete(shortName: 'remove');
+        $operation = new \Sylius\Resource\Metadata\Api\Delete(shortName: 'remove');
```
Comment (diimpp) [inline src/Component/spec/Symfony/Routing/Factory/DeleteOperationRoutePathFactorySpec.php]: Please move definition to `use` list.

File: src/Component/spec/Symfony/Routing/Factory/ShowOperationRoutePathFactorySpec.php [hunk 15]
```diff
@@ -55,7 +54,7 @@ function it_generates_route_path_for_show_operations_with_custom_short_name(): v
 
     function it_generates_route_path_for_api_get_operations(): void
     {
-        $operation = new Api\Get();
+        $operation = new \Sylius\Resource\Metadata\Api\Get();
```
Comment (diimpp) [inline src/Component/spec/Symfony/Routing/Factory/ShowOperationRoutePathFactorySpec.php]: Please move definition to `use` list.

File: src/Component/spec/Symfony/Routing/Factory/UpdateOperationRoutePathFactorySpec.php [hunk 16]
```diff
@@ -55,14 +54,14 @@ function it_generates_route_path_for_update_operations_with_custom_short_name():
 
     function it_generates_route_path_for_api_put_operations(): void
     {
-        $operation = new Api\Put();
+        $operation = new \Sylius\Resource\Metadata\Api\Put();
```
Comment (diimpp) [inline src/Component/spec/Symfony/Routing/Factory/UpdateOperationRoutePathFactorySpec.php]: Please move definition to `use` list.

File: src/Component/spec/Symfony/Routing/Factory/UpdateOperationRoutePathFactorySpec.php [hunk 17]
```diff
@@ -55,14 +54,14 @@ function it_generates_route_path_for_update_operations_with_custom_short_name():
...
 
         $this->createRoutePath($operation, '/dummies')->shouldReturn('/dummies/{id}');
     }
 
     function it_generates_route_path_for_api_patch_operations(): void
     {
-        $operation = new Api\Patch();
+        $operation = new \Sylius\Resource\Metadata\Api\Patch();
```
Comment (diimpp) [inline src/Component/spec/Symfony/Routing/Factory/UpdateOperationRoutePathFactorySpec.php]: Please move definition to `use` list.

File: src/Component/src/Metadata/Operations.php [hunk 18]
```diff
@@ -11,7 +11,7 @@
 
 declare(strict_types=1);
 
-namespace  Sylius\Component\Resource\Metadata;
+namespace  Sylius\Resource\Metadata;
```
Comment (diimpp) [inline src/Component/src/Metadata/Operations.php]:
```suggestion
namespace Sylius\Resource\Metadata;
```

File: tests/Application/src/Subscription/Entity/Subscription.php [hunk 19]
```diff
@@ -59,11 +59,11 @@
     normalizationContext: ['groups' => 'subscription:read'],
     denormalizationContext: ['groups' => 'subscription:write'],
 )]
-#[Api\GetCollection]
-#[Api\Post]
-#[Api\Put]
-#[Api\Delete]
-#[Api\Get]
+#[\Sylius\Resource\Metadata\Api\GetCollection]
```
Comment (diimpp) [inline tests/Application/src/Subscription/Entity/Subscription.php]: Please move this and bellow definitions to `use` list.

File: tests/Application/src/Subscription/Entity/Subscription.php [hunk 20]
```diff
@@ -17,16 +17,16 @@
 use App\Subscription\Twig\Context\Factory\ShowSubscriptionContextFactory;
 use Doctrine\ORM\Mapping as ORM;
 use Sylius\Component\Resource\Metadata\Api;
```
Comment (diimpp) [inline tests/Application/src/Subscription/Entity/Subscription.php]: This `use` is probably not in use anymore

File: src/Bundle/Resources/config/services/metadata.xml (hunk 10 above)
Comment (loic425) [reply src/Bundle/Resources/config/services/metadata.xml:20]: could we open an issue for that?

Comment (diimpp) [reply src/Bundle/Resources/config/services/metadata.xml:20]: Done
https://github.com/Sylius/SyliusResourceBundle/issues/776

File: src/Component/spec/Symfony/Routing/Factory/ShowOperationRoutePathFactorySpec.php (hunk 15 above)
Comment (diimpp) [reply src/Component/spec/Symfony/Routing/Factory/ShowOperationRoutePathFactorySpec.php]: @loic425 one case left

Comment (loic425) [reply src/Component/spec/Symfony/Routing/Factory/ShowOperationRoutePathFactorySpec.php]: oups sorry

Comment (loic425) [reply src/Component/spec/Symfony/Routing/Factory/ShowOperationRoutePathFactorySpec.php]: done @diimpp

**Review by diimpp — APPROVED** [review]

**Review by lchrusciel — APPROVED** [review]

**Comment by lchrusciel** [comment]
Thank you, @loic425!

**Pull request merged**